src/rebar_prv_dialyzer.erl

```diff
@@ -382,6 +384,29 @@ run_dialyzer(State, Opts) ->
             {0, State}
     end.
 
+filter_warnings([], Warnings) ->
+    Warnings;
+filter_warnings(Ignores, Warnings) ->
+    lists:filter(fun(Warning) ->
+        not lists:any(fun (Ignore) -> match_warning(Ignore, Warning) end, Ignores)
+    end, Warnings).
+
+match_warning({MatchTag, {MatchPath, MatchLine}, MatchMsg}, {Tag, {Path, Line}, Msg}) when MatchPath =/= '_' ->
+    lists:suffix(MatchPath, Path) andalso match_term([MatchTag, MatchLine, MatchMsg], [Tag, Line, Msg]);
```

Comment:
Perhaps just call `filename:basename/1` on the file in the warning itself and then pass it to `match_term/2`. I am not keen on `lists:suffix/1` nor the special casing an element in the match spec.